src/application_settings/container_base.py

```diff
@@ -159,24 +159,77 @@ def _get_saved_data(cls, throw_if_file_not_found: bool = False) -> dict[str, Any
         """Get the data stored in the parameter file"""
         data_stored: dict[str, Any] = {}
         path = cls.filepath()
-        if path is None or not path.is_file():
-            err_mess = (
-                f"Path {str(path)} not valid for {cls.kind_string().lower()} file."
-            )
-            if throw_if_file_not_found:
-                raise FileNotFoundError(err_mess)
-            print(err_mess, "Trying with defaults, but this may not work.")
-            return {}
-
-        if (ext := path.suffix[1:].lower()) == str(FileFormat.TOML.value):
-            with path.open(mode="rb") as fptr:
-                data_stored = tomllib.load(fptr)
-        elif ext == str(FileFormat.JSON.value):
-            with path.open(mode="r") as fptr:
-                data_stored = json.load(fptr)
-        else:
-            print(f"Unknown file format {ext} given in {path}.")
+        if _check_filepath(path, cls.kind_string().lower(), throw_if_file_not_found):
+            real_path = cast(Path, path)
+            if real_path.suffix[1:].lower() == str(FileFormat.TOML.value):
+                if cls.kind_string() == "Config":
+                    data_stored = _load_toml_with_includes(
+                        real_path, throw_if_file_not_found
+                    )
+                else:
+                    data_stored = _load_toml(real_path)
+            if real_path.suffix[1:].lower() == str(FileFormat.JSON.value):
+                data_stored = _load_json(real_path)
         return data_stored
 
 
+def _check_filepath(
+    path: PathOpt, file_kind: str, throw_if_file_not_found: bool
+) -> bool:
+    if path is None or not path.is_file():
+        err_mess = f"Path {str(path)} not valid for {file_kind} file."
+        if throw_if_file_not_found:
+            raise FileNotFoundError(err_mess)
+        print(err_mess, "Trying with defaults, but this may not work.")
+        return False
+    ext = path.suffix[1:].lower()
+    try:
+        FileFormat(ext)
+    except ValueError:
+        print(f"Unknown file format {ext} given in {path}.")
+        return False
+    return True
+
+
+def _load_toml(path: Path) -> dict[str, Any]:
+    data_stored: dict[str, Any] = {}
+    with path.open(mode="rb") as fptr:
+        data_stored = tomllib.load(fptr)
+    return data_stored
+
+
+def _load_json(path: Path) -> dict[str, Any]:
+    data_stored: dict[str, Any] = {}
+    with path.open(mode="r") as fptr:
+        data_stored = json.load(fptr)
+    return data_stored
+
+
+def _load_toml_with_includes(
+    path: Path, throw_if_file_not_found: bool
+) -> dict[str, Any]:
+    data_stored = _load_toml(path)
+    if included_files := data_stored.get("__include__", None):
+        if isinstance(included_files, str):
```

Comment:
I think it might be safer to do:
```
if not isinstance(included_files, list):
    included_files = [included_files]
```
That way I get a sensible raised exception when i do a:

```toml
__include__ = 5
```

Reply:
Might also be a good addition to the tests :-).

Reply:
Agreed and test added